galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/main/java/dev/galasa/zos3270/spi/Zos3270TerminalImpl.java

```diff
@@ -205,45 +217,155 @@ public synchronized void screenUpdated(Direction direction, AttentionIdentificat
+        int numRows = terminalSize.getRows() + 2;
+        int numCols = terminalSize.getColumns();
+
+        int width = numCols * 7;
+        int height = numRows * 13; 
+
+        BufferedImage image = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D graphics = image.createGraphics();
+
+        Font font = new Font(Font.MONOSPACED, Font.PLAIN, 10);
+        graphics.setFont(font);
+        FontMetrics fontMetrics = graphics.getFontMetrics();
+        int fontHeight = fontMetrics.getHeight();
+        int fontWidth = fontMetrics.getMaxAdvance();
+
+        graphics.setPaint(Color.black);
+        graphics.fillRect(0, 0, width, height);
+        graphics.setPaint(Color.green);
+
+        List<TerminalImage> terminalImages = rasTerminal.getImages();
+        for (int i = 0; i < terminalImages.size(); i++) {
+            for (TerminalField field : terminalImages.get(i).getFields()) {
+                StringBuilder sb = new StringBuilder();
+                for (FieldContents contents : field.getContents()) {
+                    int col = (field.getColumn() + 1);
+                    int row = (field.getRow() + 1);
+                    
+                    // Converting FieldContents to Strings and removing confidential text if required
+                    for (Character c : contents.getChars()) {
+                        if (c == null) {
+                            sb.append(" ");
+                        } else {
+                            sb.append(c);
+                        }
+                    }
+                    String fieldText = applyCtf ? cts.removeConfidentialText(sb.toString()) : sb.toString();
+
+                    for (Character c : fieldText.toCharArray()) {
+                        col++;
+                        if (col > numCols) {
+                            col = 1;
+                            row++;
+                            if (row > numRows) {
+                                row = 1;
+                            }
+                        }
+                        graphics.drawString(Character.toString(c), col * fontWidth, row * fontHeight);
+                    }
+                }
+            }
+
+            String terminalStatusRow = writeTerminalStatusRow(terminalImages.get(i), terminalSize.getColumns(), terminalSize.getRows());
+            graphics.drawString(terminalStatusRow, 1 * fontWidth, (terminalSize.getRows() + 1) * fontHeight);
+
+            String terminalFilename = this.terminalId + "-" + String.format("%05d", rasTerminalSequence) + "-" + (i + 1) + ".png";
+            Path terminalPath = terminalImagesDirectory.resolve(terminalFilename);
+            
+            OutputStream os = Files.newOutputStream(terminalPath,
+            new SetContentType(ResultArchiveStoreContentType.PNG),
+            StandardOpenOption.CREATE);
+
+            ImageIO.write(image, "png", os);
+            graphics.clearRect(0, 0, width, height);
```

Comment:
what benefit does this line have since the graphics object will lose reference as we exit this method

Reply:
This is at the bottom of a for loop, where each iteration we are drawing a new image, so we need to clear the rectangle of the graphics object so we can draw the next picture. We don't want to create a new graphics object here as we want to keep the colour/size settings from earlier.

Reply:
The `clearRect` call is made within the for loop that works through each image - without it, each image would be drawn on top of the previous image